src/NuGet.Core/NuGet.Configuration/Settings/SettingsLoadingContext.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.Configuration
+{
+    public sealed class SettingsLoadingContext : IDisposable
+    {
+        private readonly IList<Lazy<SettingsFile>> _settingsFiles = new List<Lazy<SettingsFile>>();
+        private readonly SemaphoreSlim _semaphore;
+        private bool _isDisposed;
+
+        public SettingsLoadingContext()
+        {
+            _semaphore = new SemaphoreSlim(initialCount: 1, maxCount: 1);
+        }
+
+        internal SettingsFile GetOrCreateSettingsFile(string filePath)
+        {
```

Comment:
check if disposed and throw `ObjectDisposedException`

Reply:
fixed